File: Backend/info-skjerm-api/Model/WeatherForecastObject.cs
namespace info_skjerm_api.Model

//Objekt oppsett basert på yr apiet
{
    public struct WeatherForecastInfo
    {
        public Properties properties { get; set; }
    }

    public struct Properties
    {
        public List<TimeSeries> timeseries { get; set; }
    }

    public struct TimeSeries
    {
        public string time { get; set; }
        public Data data { get; set; }
    }

    public struct Data
    {
        public Instant instant { get; set; }

        public Next_1_Hours next_1_hours { get; set; }
        
        public Next_6_Hours next_6_hours { get; set; }
    }

    public struct Instant
    {
        public Details details { get; set; }
    }

    public struct Details
    {
        public float air_temperature { get; set; }
    }

    public struct Next_1_Hours
    {
       public Summary summary { get; set; }
    }

    public struct Next_6_Hours
    {
        public Summary summary { get; set; }
    }

    public struct Summary
    {
        public string symbol_code { get; set; }
    }
}

File: Backend/TS-copy/Model/WeatherForecastObject.ts
export interface WeatherForecastInfo {
  properties: Properties;
}

export interface Properties {
  timeseries: TimeSeries[];
}

export interface TimeSeries {
  time: string;
  data: Data;
}

export interface Data {
  instant: Instant;
  next_1_hours: Next_1_Hours;
  next_6_hours: Next_6_Hours;
}

export interface Instant {
  details: Details;
}

export interface Details {
  air_temperature: number;
}

export interface Next_1_Hours {
  summary: Summary;
}

export interface Next_6_Hours {
  summary: Summary;
}

export interface Summary {
  symbol_code: string;
}
